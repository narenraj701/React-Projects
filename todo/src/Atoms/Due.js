import React from 'react';
import Typography from '@material-ui/core/Typography';

const Due = ({ due }) => <Typography color="textSecondary" data-testid="duetext" gutterBottom>{due}</Typography>;

export default Due;