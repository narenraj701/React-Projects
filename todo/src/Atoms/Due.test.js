import React from 'react';
import { render } from '@testing-library/react';
import {create} from 'react-test-renderer';
import Due from './Due';

test("Match the snapshot - Due message",() => {
    const input=create(<Due />);
    expect(input.toJSON()).toMatchSnapshot();
})
test("Due Text Field is present",()=>{
    const {getByTestId}=render(<Due due={"10th July"}/>);
    const element=getByTestId("duetext");
    expect(element).toBeInTheDocument();
})
test("Checking Due Meaasge Passed",()=>{
    const {getByTestId}=render(<Due due={"10th July"}/>);
    const element=getByTestId("duetext");
    expect(element).toHaveTextContent("10th July");
})

