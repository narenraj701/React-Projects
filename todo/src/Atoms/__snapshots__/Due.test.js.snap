// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - Due message 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
  data-testid="duetext"
/>
`;
